### PR TITLE
doc/dev: t5logy: rewrite "suites inventory"

### PR DESCRIPTION
**Pull request by zdover23**
This really just re-organizes this information into
a table. I think tables are easier to read than what
was here before.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by jdurgin — APPROVED** [review]
agreed, a table is more legible

File: doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro.rst [hunk 1]
```diff
@@ -96,50 +96,55 @@ Suites Inventory
...
+    - run tests that exercise the Ceph API with an actual Ceph cluster
 
-`ceph-deploy man page`_
+  * - `teuthology <https://github.com/ceph/ceph/tree/master/qa/suites/teuthology>`_ 
+    - verify that teuthology can run integration tests, with and without OpenStack
+
+  * - `upgrade <https://github.com/ceph/ceph/tree/master/qa/suites/upgrade>`_
+    - for various versions of Ceph, verify that upgrades can happen without disrupting an ongoing workload
+
+  * - `ceph-deploy man page`_
```
Comment (jdurgin) [inline doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro.rst:146]: not sure why this link is here, it should be removed (could be another pr)

**Review by neha-ojha — APPROVED** [review]
